wscript

```diff
@@ -875,6 +891,132 @@ def _load_pre_build(bld):
+                    "--sim-address=127.0.0.1",
+                    "-I0"
+                ],
+                "stopAtEntry": False,
+                "cwd": "${workspaceFolder}",
+                "environment": [],
+                "externalConsole": False,
+                "MIMode": "lldb",
+                "setupCommands": [
+                    {
+                        "description": "Enable pretty-printing for gdb",
+                        "text": "-enable-pretty-printing",
+                        "ignoreFailures": False
+                    }
+                ],
+            }
+        else:
+            if configuration['name'] == 'AP_Bootloader':
+                executable_path = "${workspaceFolder}/build/"+ cfg.options.board + "/bootloader/AP_Bootloader"
+            else:
+                executable_path = "${workspaceFolder}/build/"+ cfg.options.board + "/bin/" + configuration['target']
+            launch_configuration = {
+                "name": configuration['name'],
+                "cwd": "${workspaceFolder}",
+                "executable": executable_path,
+                "liveWatch": {
+                    "enabled": True,
+                    "samplesPerSecond": 4
+                },
+                "request": "launch",
+                "type": "cortex-debug",
+                "servertype": "openocd",
+                "configFiles": [
+                    "${workspaceFolder}/build/" + cfg.options.board + "/openocd.cfg",
+                ]
+            }
+
+        configuration_overwrited = False
+        for index, current_configuration in enumerate(launch_json["configurations"]):
+            if current_configuration.get('name') == launch_configuration['name']:
+                launch_json["configurations"][index] = launch_configuration
+                configuration_overwrited = True
+                break
+        if not configuration_overwrited:
+            launch_json["configurations"].append(launch_configuration)
+
+    with open(launch_json_path, 'w') as f:
+        json.dump(launch_json, f, indent=4)
+
+    #create openocd.cfg file
+    if cfg.options.board != 'sitl':
+        openocd_cfg_path = os.path.join(cfg.srcnode.abspath(), 'build', cfg.options.board, 'openocd.cfg')
+        mcu_type = cfg.env.get_flat('APJ_BOARD_TYPE')
+        openocd_target = ''
+        if mcu_type.startswith("STM32H7"):
+            openocd_target = 'stm32h7x.cfg'
+        elif mcu_type.startswith("STM32F7"):
+            openocd_target = 'stm32f7x.cfg'
+        elif mcu_type.startswith("STM32F4"):
+            openocd_target = 'stm32f4x.cfg'
+        elif mcu_type.startswith("STM32F3"):
+            openocd_target = 'stm32f3x.cfg'
+        elif mcu_type.startswith("STM32L4"):
+            openocd_target = 'stm32l4x.cfg'
+        elif mcu_type.startswith("STM32G4"):
+            openocd_target = 'stm32g4x.cfg'
+
+        with open(openocd_cfg_path, 'w+') as f:
+            f.write("source [find interface/stlink-v2.cfg]\n")
```

Comment:
I don't think this is right pointing to the `stlink-v2` config.

The most of the `stlink*.cfg` files point to the more common `stlink.cfg` - eg: https://github.com/openocd-org/openocd/blob/master/tcl/interface/stlink-v2.cfg

Reply:
You are right, thanks for the hint, shoule be stlink.cfg, updated